// @paciolan/primitives/dist/Card/Card.subcomponent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Footer = exports.Content = exports.Header = exports.Image = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styles = require("@material-ui/core/styles");

var _Card = require("./Card.style");

var _utilities = require("@paciolan/utilities");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propSelect = _utilities.componentUtils.propSelect;


var _Image = function _Image(props) {
  var allowedPropKeys = ["className", "children"];
  var allowedProps = propSelect(props, allowedPropKeys);
  return _react2.default.createElement("div", _extends({}, allowedProps, { "data-componentname": "card-image" }));
};

var _Header = function _Header(props) {
  var allowedPropKeys = ["className", "children"];
  var allowedProps = propSelect(props, allowedPropKeys);
  return _react2.default.createElement("div", _extends({}, allowedProps, { "data-componentname": "card-header" }));
};

var _Content = function _Content(props) {
  var allowedPropKeys = ["className", "children"];
  var allowedProps = propSelect(props, allowedPropKeys);
  return _react2.default.createElement("div", _extends({}, allowedProps, { "data-componentname": "card-content" }));
};

var _Footer = function _Footer(props) {
  var allowedPropKeys = ["className", "children"];
  var allowedProps = propSelect(props, allowedPropKeys);
  return _react2.default.createElement("div", _extends({}, allowedProps, { "data-componentname": "card-footer" }));
};

var Image = exports.Image = (0, _styles.withTheme)()((0, _Card.imageStyle)(_Image));
var Header = exports.Header = (0, _styles.withTheme)()((0, _Card.headerStyle)(_Header));
var Content = exports.Content = (0, _styles.withTheme)()((0, _Card.contentStyle)(_Content));
var Footer = exports.Footer = (0, _styles.withTheme)()((0, _Card.footerStyle)(_Footer));

Image.propTypes = {
  src: _propTypes2.default.string,
  height: _propTypes2.default.number
};

Header.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  "data-testid": _propTypes2.default.string
};

Content.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  "data-testid": _propTypes2.default.string
};

Footer.propTypes = {
  compact: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  "data-testid": _propTypes2.default.string
};
