// @paciolan/pac-theme-base/dist/theme/typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var typography = {
  headline: {
    fontSize: "1.5rem",
    fontWeight: 400,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    lineHeight: "1.35417em",
    color: "rgba(0, 0, 0, 0.87)"
  },
  title: {
    fontSize: "1.3125rem",
    fontWeight: 500,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    lineHeight: "1.16667em",
    color: "rgba(0, 0, 0, 0.87)"
  },
  subheading: {
    fontSize: "1rem",
    fontWeight: 400,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    lineHeight: "1.5em",
    color: "rgba(0, 0, 0, 0.87)"
  },
  body2: {
    fontSize: "0.875rem",
    fontWeight: 500,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    lineHeight: "1.71429em",
    color: "rgba(0, 0, 0, 0.87)"
  },
  body1: {
    fontSize: "0.875rem",
    fontWeight: 400,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    lineHeight: "1.46429em",
    color: "rgba(0, 0, 0, 0.87)"
  },
  caption: {
    fontSize: "0.75rem",
    fontWeight: 400,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    lineHeight: "1.375em",
    color: "rgba(0, 0, 0, 0.54)"
  },
  button: {
    fontSize: "0.875rem",
    textTransform: "uppercase",
    fontWeight: 500,
    fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"],
    color: "rgba(0, 0, 0, 0.87)"
  }
};

exports.default = typography;
