// @paciolan/primitives/dist/Form/Form.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.textfieldStyle = exports.checkboxStyle = exports.radioStyle = exports.radioGroupStyle = exports.fieldStyle = undefined;

var _utilities = require("@paciolan/utilities");

var styleGenerator = _utilities.componentUtils.styleGenerator;

// TODO: Use themes for size, default line-height and default margin-bottom

var style = {
  base: function base(props) {
    return "\n    margin: 0 0 " + (props.noMargin ? 0 : "24px") + ";\n  ";
  }
};

var field = {
  base: function base(props) {
    return "\n    cursor: pointer;\n    display: flex;\n    align-items: center;\n  ";
  }
};

var radioGroup = {
  base: function base(props) {
    return "\n  ";
  }
};

var radio = {
  base: function base(props) {
    return "\n  ";
  }
};

var checkbox = {
  base: function base(props) {
    return "\n  ";
  }
};

var textfield = {
  base: function base(props) {
    return "\n  ";
  }
};

var FormFieldStyle = styleGenerator(style);
var fieldStyle = exports.fieldStyle = styleGenerator(field);
var radioGroupStyle = exports.radioGroupStyle = styleGenerator(radioGroup);
var radioStyle = exports.radioStyle = styleGenerator(radio);
var checkboxStyle = exports.checkboxStyle = styleGenerator(checkbox);
var textfieldStyle = exports.textfieldStyle = styleGenerator(textfield);

exports.default = FormFieldStyle;
