// @paciolan/utilities/dist/theme/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(["\n    // global styles\n    body {\n      margin: 0;\n      padding: 0;\n      font-size: 100%;\n      line-height: 1;\n    }\n\n    div.tooltip-popper {\n      padding: 6px 8px;\n      width: 226px;\n    }\n\n    div.tooltip-popper > div {\n      font-size: 10px;\n      width: 198px;\n      text-align: left;\n    }\n  "], ["\n    // global styles\n    body {\n      margin: 0;\n      padding: 0;\n      font-size: 100%;\n      line-height: 1;\n    }\n\n    div.tooltip-popper {\n      padding: 6px 8px;\n      width: 226px;\n    }\n\n    div.tooltip-popper > div {\n      font-size: 10px;\n      width: 198px;\n      text-align: left;\n    }\n  "]),
    _templateObject2 = _taggedTemplateLiteral(["\n    // app styles\n    display: flex;\n    flex-direction: column;\n    min-height: 100vh;\n    font-family: \"Roboto\", \"Helvetica\", \"Arial\", \"sans-serif\";\n    padding: 30px;\n  "], ["\n    // app styles\n    display: flex;\n    flex-direction: column;\n    min-height: 100vh;\n    font-family: \"Roboto\", \"Helvetica\", \"Arial\", \"sans-serif\";\n    padding: 30px;\n  "]);

var _pacThemeBase = require("@paciolan/pac-theme-base");

var _pacThemeBase2 = _interopRequireDefault(_pacThemeBase);

var _styledComponents = require("styled-components");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

exports.default = (0, _pacThemeBase.withTheme)(_pacThemeBase2.default).extendInto({
  global: (0, _styledComponents.css)(_templateObject),
  app: (0, _styledComponents.css)(_templateObject2),
  typography: {
    body1: {
      fontFamily: ["Roboto", "Helvetica", "Arial", "sans-serif"]
    },
    columnHeaderText: {
      display: "inline-block"
    },
    ellipsis: {
      whiteSpace: "normal",
      overflow: "visible",
      textOverflow: "ellipsis",
      maxWidth: "100%",
      width: "100%"
    },
    flex: {
      display: "flex",
      row: "row",
      column: "column",
      start: "flex-start",
      end: "flex-end",
      center: "center"
    },
    opacity: {
      0: "0",
      1: "0.1",
      2: "0.2",
      3: "0.3",
      4: "0.4",
      5: "0.5",
      6: "0.6",
      7: "0.7",
      8: "0.8",
      9: "0.9",
      10: "1"
    },
    gridHeader: {
      whiteSpace: "normal",
      overflow: "visible",
      textOverflow: "ellipsis",
      maxWidth: "100%",
      display: "flex",
      fontSize: "0.75rem",
      fontWeight: "400"
    }
  },
  palette: {
    // defined by UX team, notify if changed
    gray: {
      gray1: "#f3f3f3",
      gray2: "#e6e6e6",
      gray3: "#d7d7d7",
      gray4: "#9a9a9a",
      gray5: "#6a6a6a",
      gray6: "#333333"
    }
  }
});
