// @paciolan/primitives/dist/Button/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Button2 = require("@material-ui/core/Button");

var _Button3 = _interopRequireDefault(_Button2);

var _styles = require("@material-ui/core/styles");

var _utilities = require("@paciolan/utilities");

var _Button4 = require("./Button.style");

var _Button5 = _interopRequireDefault(_Button4);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propSelect = _utilities.componentUtils.propSelect;


var _Button = function _Button(props) {
  var allowedPropKeys = ["disabled", "fullWidth", "href", "onClick", "size", "children", "className"];
  var allowedProps = propSelect(props, allowedPropKeys);

  if (props.theme.context === "operator") {
    allowedProps = _extends({}, allowedProps, {
      disableRipple: true
    });
  }

  return _react2.default.createElement(_Button3.default, _extends({}, allowedProps, { "data-componentname": "button" }));
};

var Button = (0, _Button5.default)(_Button);

exports.default = (0, _styles.withTheme)()(Button);


Button.muiName = "Button";

Button.propTypes = {
  onClick: _propTypes2.default.func,
  children: _propTypes2.default.node,
  bold: _propTypes2.default.bool,
  theme: _propTypes2.default.object,
  disabled: _propTypes2.default.bool,
  fullWidth: _propTypes2.default.bool,
  href: _propTypes2.default.string,
  className: _propTypes2.default.string,
  "data-testid": _propTypes2.default.string
};
