// @paciolan/utilities/dist/storybook/storybookDecorators.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.backgroundDecorator = exports.themeDecorator = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _ThemeProvider = require("../theme/ThemeProvider");

var _ThemeProvider2 = _interopRequireDefault(_ThemeProvider);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var themeDecorator = exports.themeDecorator = function themeDecorator(theme) {
  return function (storyFn) {
    return _react2.default.createElement(
      _ThemeProvider2.default,
      { theme: theme },
      _react2.default.createElement(
        "div",
        null,
        storyFn()
      )
    );
  };
};

var backgroundDecorator = exports.backgroundDecorator = function backgroundDecorator(backgroundColor) {
  return function (storyFn) {
    return _react2.default.createElement(
      "div",
      {
        style: {
          backgroundColor: backgroundColor,
          position: "absolute",
          top: 0,
          bottom: 0,
          right: 0,
          left: 0,
          padding: 20,
          overflow: "auto"
        }
      },
      storyFn()
    );
  };
};
