// @paciolan/primitives/dist/Link/Link.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styleWrapper = exports.styleLink = undefined;

var _utilities = require("@paciolan/utilities");

var _LinkConsumer = require("./Link.consumer.style");

var _LinkConsumer2 = _interopRequireDefault(_LinkConsumer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styleGenerator = _utilities.componentUtils.styleGenerator;


var style = {
  link: {
    base: function base(props) {
      return "\n      font-family: " + props.theme.typography.fontFamily + ";\n      font-size: 12px;\n      color: #23e;\n      font-weight: " + (props.bold ? 700 : 400) + ";\n      text-decoration: underline;\n      line-height: 1em;\n      margin: 0 " + (!props.noMargin ? "0.2em 1em" : "0 0") + ";\n      &:hover{\n        font-color: #12b;\n        text-decoration: none;\n        cursor: pointer;\n      }\n      &:visited{\n        color: #23e;\n      }\n    ";
    },
    consumer: _LinkConsumer2.default
  },
  wrapper: {
    base: function base(props) {
      return "\n      display: " + (props.align ? "block" : "inline") + ";\n      text-align: " + (props.align || "initial") + ";\n    ";
    }
  }
};

var styleLink = exports.styleLink = styleGenerator(style.link);
var styleWrapper = exports.styleWrapper = styleGenerator(style.wrapper);
