// @paciolan/primitives/dist/Text/Text.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utilities = require("@paciolan/utilities");

var _TextConsumer = require("./Text.consumer.styles");

var _TextConsumer2 = _interopRequireDefault(_TextConsumer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styleGenerator = _utilities.componentUtils.styleGenerator;


var size = {
  smaller: 9,
  small: 11,
  default: 14,
  large: 16,
  larger: 18
};

// TODO: Use themes for size, default line-height and default margin-bottom
var style = {
  base: function base(props) {
    return "\n    text-align: " + (props.align || "") + ";\n    font-family: " + props.theme.typography.fontFamily + ";\n    font-weight: " + (props.bold ? 700 : 400) + ";\n    font-size: " + (size[props.size] || size.default) + "px;\n    line-height: " + (props.paragraph ? "1.55em" : "1em") + ";\n    margin: 0 0 " + (props.paragraph || !props.noMargin ? "0.8em" : 0) + ";\n  ";
  },
  consumer: _TextConsumer2.default
};

var TextStyle = styleGenerator(style);

exports.default = TextStyle;
