// @paciolan/primitives/dist/Text/Text.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styles = require("@material-ui/core/styles");

var _Text2 = require("./Text.style");

var _Text3 = _interopRequireDefault(_Text2);

var _utilities = require("@paciolan/utilities");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propSelect = _utilities.componentUtils.propSelect;


var _Text = function _Text(props) {
  var allowedPropKeys = ["children", "className"];
  var allowedProps = propSelect(props, allowedPropKeys);

  return _react2.default.createElement("p", _extends({}, allowedProps, { "data-componentname": "text" }));
};

var Text = (0, _Text3.default)(_Text);

exports.default = (0, _styles.withTheme)()(Text);


Text.propTypes = {
  children: _propTypes2.default.node,
  size: _propTypes2.default.oneOf(["smaller", "small", "default", "large", "larger"]),
  align: _propTypes2.default.oneOf(["left", "center", "right", "justify"]),
  bold: _propTypes2.default.bool,
  noMargin: _propTypes2.default.bool,
  paragraph: _propTypes2.default.bool,
  "data-testid": _propTypes2.default.string
};

Text.defaultProps = {
  size: "default"
};
