// @paciolan/primitives/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Button = require("./Button");

Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Button).default;
  }
});

var _Text = require("./Text");

Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Text).default;
  }
});

var _Form = require("./Form");

Object.defineProperty(exports, "Form", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Form).default;
  }
});

var _Link = require("./Link");

Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Link).default;
  }
});

var _Heading = require("./Heading");

Object.defineProperty(exports, "Heading", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Heading).default;
  }
});

var _Card = require("./Card");

Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Card).default;
  }
});

var _Image = require("./Image");

Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Image).default;
  }
});

var _Span = require("./Span");

Object.defineProperty(exports, "Span", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Span).default;
  }
});

var _Layout = require("./Layout");

Object.defineProperty(exports, "Layout", {
  enumerable: true,
  get: function get() {
    return _Layout.Layout;
  }
});
Object.defineProperty(exports, "Row", {
  enumerable: true,
  get: function get() {
    return _Layout.Row;
  }
});
Object.defineProperty(exports, "Column", {
  enumerable: true,
  get: function get() {
    return _Layout.Column;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
