// @paciolan/primitives/dist/Heading/Heading.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utilities = require("@paciolan/utilities");

var _HeadingConsumer = require("./Heading.consumer.styles");

var _HeadingConsumer2 = _interopRequireDefault(_HeadingConsumer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styleGenerator = _utilities.componentUtils.styleGenerator;


var size = {
  h1: 32,
  h2: 24,
  h3: 18,
  h4: 16,
  h5: 13,
  h6: 11
};

// TODO: Use themes for size, default line-height and default margin-bottom
var style = {
  base: function base(props) {
    return "\n    color: #333;\n    text-align: " + (props.align || "") + ";\n    text-decoration: none;\n    font-family: " + props.theme.typography.fontFamily + ";\n    font-weight: " + (props.bold ? 700 : 400) + ";\n    font-size: " + size[props.htmlElement] + "px;\n    line-height: " + (props.lineHeight ? props.lineHeight + "px" : "1.3em") + ";\n    margin-top: 0;\n    margin-bottom: " + (props.noMargin ? 0 : "0.8em") + ";\n  ";
  },
  consumer: _HeadingConsumer2.default
};

var HeadingStyle = styleGenerator(style);

exports.default = HeadingStyle;
