// @paciolan/primitives/dist/Heading/Heading.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styles = require("@material-ui/core/styles");

var _Heading2 = require("./Heading.style");

var _Heading3 = _interopRequireDefault(_Heading2);

var _utilities = require("@paciolan/utilities");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propSelect = _utilities.componentUtils.propSelect;


var _Heading = function _Heading(props) {
  var allowedPropKeys = ["children", "className"];
  var allowedProps = _extends({}, propSelect(props, allowedPropKeys), {
    "data-componentname": "heading"
  });

  return _react2.default.createElement(props.htmlElement, allowedProps, props.children);
};

var Heading = (0, _Heading3.default)(_Heading);

exports.default = (0, _styles.withTheme)()(Heading);


Heading.propTypes = {
  children: _propTypes2.default.node,
  align: _propTypes2.default.oneOf(["left", "center", "right"]),
  bold: _propTypes2.default.bool,
  htmlElement: _propTypes2.default.oneOf(["h1", "h2", "h3", "h4", "h5", "h6"]),
  lineHeight: _propTypes2.default.number,
  noMargin: _propTypes2.default.bool,
  "data-testid": _propTypes2.default.string
};

Heading.defaultProps = {
  htmlElement: "h1"
};
